packages/manager/modules/pci/src/projects/project/instances/instance/shelve/shelve.controller.js

```diff
@@ -0,0 +1,76 @@
+import get from 'lodash/get';
+
+export default class PciInstanceShelveController {
+  /* @ngInject */
+  constructor(
+    $q,
+    $translate,
+    OvhApiCloudProjectInstance,
+    PciProjectsProjectInstanceService,
+    Poller,
+  ) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.OvhApiCloudProjectInstance = OvhApiCloudProjectInstance;
+    this.PciProjectsProjectInstanceService = PciProjectsProjectInstanceService;
+    this.Poller = Poller;
+  }
+
+  $onInit() {
+    this.isLoading = false;
+  }
+
+  startPollerForShelveInstance() {
+    const endPointUrl = `/cloud/project/${this.projectId}/instance/${this.instance.id}`;
+    this.Poller.poll(endPointUrl, null, {
+      interval: 3000,
+      successRule(instance) {
+        return instance.status === 'SHELVED_OFFLOADED';
+      },
+      namespace: 'cloud.project.instance.shelve',
+      notifyOnError: false,
+    }).then(() => {
+      if (this.isInstancesStateScope()) {
+        return this.refreshInstances();
+      }
+
+      return this.$q.when();
```

Comment:
useless

Reply:
Yes true that is useless @frenautvh, but i implemented to preserve promises strategy.
if it's bad practice or insecure implementation ask me to remove them.

Reply:
@mohammed-zahaf
when you are in a 'then' block of a promise, if you return nothing the promise chain won't be broken, it's useless to return $q.when() ; it's the same thing as 'return undefined' or just nothing

Reply:
Thanks